packages/driver/cypress/fixtures/auth/idp.html

```diff
@@ -6,40 +6,62 @@
   <label for="name">Username: </label>
   <input data-cy="username" type="text" name="name" id="name" required>
   <button data-cy="login">Login</button>
+  <button data-cy="slow-login">Slow Login</button>
 
   <script>
-    document.querySelector('[data-cy="login"]').addEventListener('click', () => {
-
+    const createJwt = (username) => {
       const expiry = new Date()
       // Expire in 10 minutes
       expiry.setMinutes(expiry.getMinutes() + 10)
       // create faux jwt
-      const fauxJwt = {
+      return {
         header: {
           alg: 'string',
           type: 'JWT'
         },
         body: {
-          username: document.querySelector('[data-cy="username"]').value,
+          username,
           expiry: expiry.toISOString()
         }
       }
-      // redirect
+    }
+
+    const getRedirectHref = () => {
       const queryString = window.location.search;
       const urlParams = new URLSearchParams(queryString);
       let redirect = urlParams.get('redirect')
 
       if (redirect){
-        redirect = decodeURIComponent(redirect)
-      } else {
-        redirect = 'http://localhost:3500/fixtures/auth/index.html'
+        return decodeURIComponent(redirect)
       }
 
-      const redirectUrl = new URL(redirect)
+      return 'http://localhost:3500/fixtures/auth/index.html'
+    }
+
+    const addTokenToRedirect = (href, jwt) => {
+      const redirectUrl = new URL(href)
       const redirectUrlParams = new URLSearchParams(redirectUrl.search);
-      redirectUrlParams.append('token', encodeURIComponent(JSON.stringify(fauxJwt)))
+      redirectUrlParams.append('token', encodeURIComponent(JSON.stringify(jwt)))
       redirectUrl.search = redirectUrlParams.toString()
-      window.location.href = redirectUrl.href
+      return redirectUrl.href
+    }
+
+    document.querySelector('[data-cy="login"]').addEventListener('click', () => {
+      const jwt = createJwt(document.querySelector('[data-cy="username"]').value)
+
+      const redirectHref = getRedirectHref()
+
+      window.location.href = addTokenToRedirect(redirectHref, jwt)
+    })
+
+    document.querySelector('[data-cy="login"]').addEventListener('click', () => {
```

Comment:
I think this is supposed to be the slow log in example?
```suggestion
    document.querySelector('[data-cy="slow-login"]').addEventListener('click', () => {
```

Reply:
yes!